blog/2021/apisix.adoc

```diff
@@ -0,0 +1,184 @@
+:title: How to Integrate Keycloak for Authentication with Apache APISIX
+:date: 2021-12-13
+:publish: true
+:author: Xinxin Zhu & Yilin Zeng
+
+image::${blogImages}/apisix/keycloak-apisix-integration.png[keycloak-apisix-integration]
+
+This article shows you how to use OpenID-Connect protocol and Keycloak for identity authentication in Apache APISIX through detailed steps.
+
+https://www.keycloak.org/[Keycloak] is an open source identity and access management solution for modern applications and services. Keycloak supports Single-Sign On, which enables services to interface with Keycloak through protocols such as OpenID Connect, OAuth 2.0, etc. Keycloak also supports integrations with different authentication services, such as Github, Google and Facebook.
+
+In addition, Keycloak also supports user federation, and can import users through LDAP and Kerberos. For more information about Keycloak, please refer to the https://www.keycloak.org/about[official documentation].
+
+== How to use
+
+=== Environment preparation
+Make sure that Apache APISIX is started in your environment before proceeding with the following steps.
```

Comment:
How? ;)

Reply:
Add steps of installing and starting APISIX.